src/states/node.rs

```diff
@@ -1593,6 +1616,72 @@ impl Node {
         result.with_value(Ok(()))
     }
 
+    fn handle_rt_req(&mut self,
+                     msg_id: MessageId,
+                     digest: sha256::Digest,
+                     src: Authority<XorName>,
+                     dst: Authority<XorName>)
+                     -> Evented<Result<(), RoutingError>> {
+        // TODO: Compare the hash and don't send anything if he requester is up-to-date.
+        let mut result = Evented::empty();
+        let sections = self.peer_mgr.pub_ids_by_section(self.full_id.public_id());
+        let serialised_sections = try_ev!(serialisation::serialise(&sections), result);
+        if digest == sha256::hash(&serialised_sections) {
+            return result.with_value(Ok(()));
+        }
+        for (prefix, members) in sections {
+            let content = MessageContent::RoutingTableResponse {
+                message_id: msg_id,
+                prefix: prefix,
+                members: members,
+            };
+            let request_msg = RoutingMessage {
+                src: dst,
+                dst: src,
+                content: content,
+            };
+            if let Err(err) = self.send_routing_message(request_msg).extract(&mut result) {
+                debug!("{:?} Failed to send RoutingTableRequest: {:?}.", self, err);
+            }
+        }
+        result.with_value(Ok(()))
+    }
+
+    fn handle_rt_rsp(&mut self,
+                     prefix: Prefix<XorName>,
+                     members: BTreeSet<PublicId>,
+                     message_id: MessageId)
+                     -> Evented<Result<(), RoutingError>> {
+        let mut result = Evented::empty();
+        if Some(message_id) != self.rt_msg_id {
```

Comment:
Shouldn't the `if` here check whether the `message_id` is _equal_ to what we sent the last time?